subsys/bluetooth/audio/ascs.c

```diff
@@ -1636,12 +1629,10 @@ static ssize_t ascs_config(struct bt_ascs *ascs, struct net_buf_simple *buf)
 
 void bt_ascs_foreach_ep(struct bt_conn *conn, bt_bap_ep_func_t func, void *user_data)
 {
-	sys_snode_t *ase_node;
 	struct bt_ascs *ascs = ascs_get(conn);
+	struct bt_ascs_ase *ase, *tmp;
 
-	SYS_SLIST_FOR_EACH_NODE(&ascs->ases, ase_node) {
-		struct bt_ascs_ase *ase = CONTAINER_OF(ase_node, struct bt_ascs_ase, node);
-
+	SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&ascs->ases, ase, tmp, node) {
```

Comment:
I assume that the _SAFE version is chosen here in case that the callback removes an ASE?

Reply:
As it's used inside of `bt_ascs_foreach_ep`, one cannot say that it won't be used to remove the entries from the list - eg. upper layer will call `bt_bap_unicast_server_release` from `bt_bap_ep_func_t` context. Thus it's better to use _SAFE version of the iterator.